firebase-common/src/main/java/com/google/firebase/heartbeatinfo/DefaultHeartBeatInfo.java

```diff
@@ -20,20 +20,27 @@
 import androidx.annotation.VisibleForTesting;
 import com.google.firebase.components.Component;
 import com.google.firebase.components.Dependency;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
 
 /** Provides information as whether to send heart beat or not. */
 public class DefaultHeartBeatInfo implements HeartBeatInfo {
 
   private HeartBeatInfoStorage storage;
 
-  private DefaultHeartBeatInfo(Context context) {
+  private boolean doesSourceExist;
+
+  private DefaultHeartBeatInfo(Context context, Set<HeartBeatLogSource> logSources) {
     storage = HeartBeatInfoStorage.getInstance(context);
+    doesSourceExist = (logSources.size() > 0);
```

Comment:
you may want to hold on to the set and check if it's empty every time, this will make it future proof for when we add support for dynamic loading.